automation_tools/__init__.py

```diff
@@ -388,17 +388,13 @@ def setup_firewall(definitions=None, flush=True):
         if run('systemctl status firewalld', quiet=True).failed:
             run('systemctl enable firewalld')
             run('systemctl start firewalld')
-        exists_command = 'firewall-cmd --query-port="{1}/{0}"'
+        exists_command = 'firewall-cmd --permanent --query-port="{1}/{0}"'
```

Comment:
checking for existing rule is done in permanent rules list as we flush the rules from permanent only 
( in runtime rules they still exist so it would skip their creation incorrectly )